birdhouse/birdhouse-compose.sh

```diff
@@ -135,53 +158,40 @@ fi
 
 log INFO "Executing docker-compose with extra options: $* ${COMPOSE_EXTRA_OPTS}"
 # the PROXY_HTTP_PORT is a little trick to make the compose file invalid without the usage of this wrapper script
-PROXY_HTTP_PORT=80 HOSTNAME=${BIRDHOUSE_FQDN} ${DOCKER_COMPOSE} ${COMPOSE_CONF_LIST} $* ${COMPOSE_EXTRA_OPTS}
-ERR=$?
-if [ ${ERR} -gt 0 ]; then
-  log ERROR "docker-compose error, exit code ${ERR}"
-  exit ${ERR}
-fi
+PROXY_HTTP_PORT=80 HOSTNAME=${BIRDHOUSE_FQDN} ${DOCKER_COMPOSE} ${COMPOSE_CONF_LIST} $* ${COMPOSE_EXTRA_OPTS} || expect_exit
 
 # execute post-compose function if exists and no error occurred
-type post-compose 2>&1 | grep 'post-compose is a function' > /dev/null
-if [ $? -eq 0 ]
+if type post-compose 2>&1 | grep 'post-compose is a function' > /dev/null
 then
-  [ ${ERR} -gt 0 ] && { log ERROR "Error occurred with docker-compose, not running post-compose"; exit $?; }
```

Comment:
Depending on the use case of `set +e; set -x`, might need to leave that in.

Reply:
Perso I have never had the need for `+e` but that's probably because `-e` was never enforced !

So how about flipping this upside down and have an escape route to not force `-e` so then we do not really need to support `+e`?

Reply:
This (https://github.com/bird-house/birdhouse-deploy/pull/519#discussion_r2018773906) could solve that problem as well I think

`+e` would be the same as setting `BIRDHOUSE_FAIL_FAST=false`

Reply:
I support using `BIRDHOUSE_FAIL_FAST=true` by default. It seems to be the most reasonable default. If disabled explicitly, things can break and the user is aware of it since they disabled it.

In the specific case here, it might still be relevant to have some "checkpoint logs", even if they do not fail the execution. Of course, we cannot have all errors logged since there's an infinity of combinations, but the main pre/"core"/post-compose steps seems relevant at least to narrow down to which step one should investigate where errors happen. 
If not enough of a concern, I am fine to leave it as is (removing the log).